plugins/app/src/main/java/com/mapbox/mapboxsdk/plugins/testapp/activity/LocalizationPlugin.java

```diff
@@ -0,0 +1,114 @@
+package com.mapbox.mapboxsdk.plugins.testapp.activity;
+
+import android.support.annotation.NonNull;
+
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.style.layers.Layer;
+import com.mapbox.mapboxsdk.style.layers.PropertyFactory;
+import com.mapbox.mapboxsdk.style.layers.SymbolLayer;
+
+import java.util.List;
+import java.util.Locale;
+
+/**
+ * The localization plugin enables localization of map labels into the user’s preferred language
+ * <p>
+ * Initialise this plugin in the {@link com.mapbox.mapboxsdk.maps.OnMapReadyCallback#onMapReady(MapboxMap)} and provide
+ * a valid instance of a {@link MapboxMap}.
+ * </p>
+ */
+public final class LocalizationPlugin {
+
+  private List<Layer> listOfMapLayers;
+
+  /**
+   * Create a localization plugin.
+   *
+   * @param mapboxMap the MapboxMap to apply localization plugin with
+   * @since 0.1.0
+   */
+  public LocalizationPlugin(@NonNull final MapboxMap mapboxMap) {
+    listOfMapLayers = mapboxMap.getLayers();
+    String deviceLanguage = Locale.getDefault().getLanguage();
+    for (Layer layer : listOfMapLayers) {
+      if (layer instanceof SymbolLayer) {
+        layer.setProperties(PropertyFactory.textField(String.format("{name_%s}", deviceLanguage)));
```

Comment:
What if the symbol layer had no `textField` to begin with, or what if it was set to `{ref}`? Would highway shields start displaying road names instead of route numbers?

Reply:
It can definitely be cleaned up, but here's my`textField` check for now 😕 : https://github.com/mapbox/mapbox-plugins-android/blob/5a1c2854a6935d4ab0ae9a3eb8ede0a65680a7bd/plugin-localization/src/main/java/com.mapbox.mapboxsdk.plugins.localization/LocalizationPlugin.java#L36-L42

`{abbr}` check and adjustment: https://github.com/mapbox/mapbox-plugins-android/blob/5a1c2854a6935d4ab0ae9a3eb8ede0a65680a7bd/plugin-localization/src/main/java/com.mapbox.mapboxsdk.plugins.localization/LocalizationPlugin.java#L43-L46